doc/source/whatsnew/v0.23.1.txt

```diff
@@ -88,6 +88,7 @@ Indexing
 - Bug in :meth:`MultiIndex.set_names` where error raised for a ``MultiIndex`` with ``nlevels == 1`` (:issue:`21149`)
 - Bug in :class:`IntervalIndex` constructors where creating an ``IntervalIndex`` from categorical data was not fully supported (:issue:`21243`, issue:`21253`)
 - Bug in :meth:`MultiIndex.sort_index` which was not guaranteed to sort correctly with ``level=1``; this was also causing data misalignment in particular :meth:`DataFrame.stack` operations (:issue:`20994`, :issue:`20945`, :issue:`21052`)
+- Bug in :func:`pandas.core.arrays.categorical._factorize_from_iterable` inappropriately caused ``TypeError`` to be raised when an ``Enum`` was used as a factor in a ``MultiIndex`` (:issue:`21298`)
```

Comment:
> @gfyoung I'm stumped by this whatsnew conflict, any thoughts?

@benediamond : I would actually move your `whatsnew` addition to `0.24.0`.   That will suffice. 😄

Reply:
I've gone ahead and just removed it---I'm not sure where it belongs, as it appeared that this PR (when it was passing) was slated for the next milestone. It was: "Bug in :func:`pandas.core.arrays.categorical._factorize_from_iterable` inappropriately caused ``TypeError`` to be raised when an ``Enum`` was used as a factor in a ``MultiIndex`` (:issue:`21298`)"

Reply:
@benediamond : I would put it in `0.24.0`, as I suggested above.  We will need a `whatsnew` for this for sure, and that would be the best place to put it.

Reply:
@gfyoung Done. this one didn't create merge conflicts, thankfully